gmail_mcp_server/gmail_client.py

```
@@ -128,7 +128,37 @@ def list_unread_emails(self, subject_filter: str | None = None, max_results: int
             return emails
 
         except HttpError as error:
-            raise Exception(f"An error occurred while listing emails: {error}")
+            raise Exception(f"An error occurred while searching emails: {error}")
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)